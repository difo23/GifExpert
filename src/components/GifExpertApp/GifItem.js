import React from 'react';
import PropTypes from 'prop-types';

const GifItem = ({ key, title, url }) => {
	return (
		<li key={key} className="card animate__animated animate__fadeIn">
			<h4>{title}</h4>
			<img src={url} alt={title} />
		</li>
	);
};

GifItem.propTypes = {
	title: PropTypes.string,
	url: PropTypes.string
};

export default GifItem;
