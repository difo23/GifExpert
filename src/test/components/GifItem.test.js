import React from 'react';
import GifItem from '../../components/GifExpertApp/GifItem';
import { shallow } from 'enzyme';

describe('Test GifItem Component', () => {
	const id = '123456';
	const url = 'http://localhost:3000/algo.png';
	const title = 'imagen de pruewba';
	const alt = title;
	const wrapper = shallow(<GifItem key={id} url={url} title={title} />);

	test('Test prueba estructra GifItem', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test(`Debe poseer un title ${title}`, () => {
		const h4 = wrapper.find('h4');

		expect(h4.text().trim()).toBe(title);
	});

	test(`Debe poseer un url ${url} y alt ${alt}`, () => {
		const img = wrapper.find('img');

		expect(img.prop('alt')).toBe(alt);
		expect(img.prop('src')).toBe(url);
	});

	test(`Debe poseer un fadein animated`, () => {
		const div = wrapper.find('div');
		const className = div.prop('className');

		expect(className.includes('animate__fadeIn')).toBe(true);
	});
});
