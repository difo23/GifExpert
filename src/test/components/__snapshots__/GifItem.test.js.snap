// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItem Component Test prueba estructra GifItem 1`] = `
<div
  className="card animate__animated animate__fadeIn"
  key="123456"
>
  <h4>
    imagen de pruewba
  </h4>
  <img
    alt="imagen de pruewba"
    src="http://localhost:3000/algo.png"
  />
</div>
`;
